src/core/container.jl

```diff
@@ -212,11 +216,109 @@ function resample!(
 
     # replace particles and log weights in the container with new particles and weights
     pc.vals = children
-    pc.logWs = zeros(n)
+    reset_logweights!(pc)
 
     pc
 end
 
+"""
+    reweight!(pc::ParticleContainer)
+
+Check if the final time step is reached, and otherwise reweight the particles by
+considering the next observation.
+"""
+function reweight!(pc::ParticleContainer)
+    n = length(pc)
+
+    particles = collect(pc)
+    numdone = 0
+    for i in 1:n
+        p = particles[i]
+
+        # Obtain ``\\log p(yₜ | y₁, …, yₜ₋₁, x₁, …, xₜ, θ₁, …, θₜ)``, or `nothing` if the
+        # the execution of the model is finished.
+        # Here ``yᵢ`` are observations, ``xᵢ`` variables of the particle filter, and
+        # ``θᵢ`` are variables of other samplers.
+        score = Libtask.consume(p)
+
+        if score === nothing
+            numdone += 1
+        else
+            # Increase the unnormalized logarithmic weights, accounting for the variables
+            # of other samplers.
+            increase_logweight!(pc, i, score + getlogp(p.vi))
```

Comment:
As a side note, `getlotp(p.vi)` will always return `0`, since the assume and observe functions for particle samplers does not modify `vi.logp` by default. This doesn't affect correctness, but worth to pay attention. 

See:
- https://github.com/TuringLang/Turing.jl/blob/v0.9.0/src/inference/AdvancedSMC.jl#L288
- https://github.com/TuringLang/Turing.jl/blob/v0.9.0/src/inference/AdvancedSMC.jl#L293

Reply:
That's not completely true (or maybe I misunderstand you), in this line `getlogp` can actually return nonzero values due to https://github.com/TuringLang/Turing.jl/blob/35e3fe8e356212ba4e64265a870833e3a9b9f4e0/src/inference/AdvancedSMC.jl#L286. However, since we call `resetlogp!` in one of the following lines, this won't show up in the saved transitions.

Reply:
I must have missed that line, thanks for the pointer!